core/src/main/java/com/linecorp/armeria/server/PathMapping.java

```diff
@@ -15,166 +15,26 @@
-    }
-
-    /**
-     * Creates a new {@link PathMapping} that matches a {@linkplain ServiceRequestContext#path() path}
-     * under the specified directory prefix. It also removes the specified directory prefix from the matched
-     * path so that {@link ServiceRequestContext#path()} does not have the specified directory prefix.
-     * For example, when {@code pathPrefix} is {@code "/foo/"}:
-     * <ul>
-     *   <li>{@code "/foo/"} translates to {@code "/"}</li>
-     *   <li>{@code "/foo/bar"} translates to  {@code "/bar"}</li>
-     *   <li>{@code "/foo/bar/baz"} translates to {@code "/bar/baz"}</li>
-     * </ul>
-     * This method is a shortcut to {@link #ofPrefix(String, boolean) ofPrefix(pathPrefix, true)}.
-     */
-    static PathMapping ofPrefix(String pathPrefix) {
-        return ofPrefix(pathPrefix, true);
-    }
-
-    /**
-     * Creates a new {@link PathMapping} that matches a {@linkplain ServiceRequestContext#path() path}
-     * under the specified directory prefix. When {@code stripPrefix} is {@code true}, it also removes the
-     * specified directory prefix from the matched path so that {@link ServiceRequestContext#path()}
-     * does not have the specified directory prefix. For example, when {@code pathPrefix} is {@code "/foo/"}:
-     * <ul>
-     *   <li>{@code "/foo/"} translates to {@code "/"}</li>
-     *   <li>{@code "/foo/bar"} translates to  {@code "/bar"}</li>
-     *   <li>{@code "/foo/bar/baz"} translates to {@code "/bar/baz"}</li>
-     * </ul>
-     */
-    static PathMapping ofPrefix(String pathPrefix, boolean stripPrefix) {
-        requireNonNull(pathPrefix, "pathPrefix");
-        if ("/".equals(pathPrefix)) {
-            // Every path starts with '/'.
-            return ofCatchAll();
-        }
-
-        return new PrefixPathMapping(pathPrefix, stripPrefix);
-    }
-
-    /**
-     * Creates a new {@link PathMapping} that performs an exact match.
-     */
-    static PathMapping ofExact(String exactPath) {
-        return new ExactPathMapping(exactPath);
-    }
-
-    /**
-     * Returns a singleton {@link PathMapping} that always matches any path.
-     */
-    static PathMapping ofCatchAll() {
-        return CatchAllPathMapping.INSTANCE;
-    }
+interface PathMapping {
 
     /**
      * Matches the specified {@code path} and extracts the path parameters from it.
      *
-     * @param mappingCtx a context to find the {@link Service}.
+     * @param routingCtx a context to find the {@link Service}.
      *
-     * @return a non-empty {@link PathMappingResult} if the specified {@code path} matches this mapping.
-     *         {@link PathMappingResult#empty()} if not matches.
+     * @return a settable {@link RoutingResultBuilder} if the specified {@code path} matches this mapping.
+     *         {@link RoutingResult#immutableBuilder()} if not matches.
      */
-    PathMappingResult apply(PathMappingContext mappingCtx);
+    RoutingResultBuilder apply(RoutingContext routingCtx);
```

Comment:
Just curious. For not-matched case, would `@Nullable` and returning `null` be better choice?

Reply:
Ah, yeah, I don't know why I couldn't think of it. 😅 Will change it. Thanks!